src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,28 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+
+    private static final int MAX_VALUE = 5;
+
+    private static final int VALUE = 10;
+
+    private final Random random = new Random();
+
+    private final ColorSupplier colorSupplier;
+
+    public FigureSupplier(ColorSupplier colorSupplier) {
+        this.colorSupplier = colorSupplier;
+    }
+
+    public Figure getRandomFigure() {
+        return new Rectangle(colorSupplier.getRandomColor(),
+                random.nextInt(MAX_VALUE),
+                random.nextInt(MAX_VALUE));
+    }

```

Comment:
This method should return random type of figure, not just rectangle.
You need additional variable that will help you to also randomize figure type:
```suggestion
    public Figure getRandomFigure() {
        int figureNumber = random.nextInt(FIGURES_COUNT);
        switch (figureNumber) {
            case 0:
                return new Rectangle(colorSupplier.getRandomColor(),
                    random.nextInt(MAX_VALUE),
                    random.nextInt(MAX_VALUE));
            case 1:
                return new Square(.....
    }

```